index.bs

```diff
@@ -990,6 +990,174 @@ while (true) {
+  if (mostRecentEntry != null &&
+      mostRecentEntry.getBoundingClientRect().top <
+          log.getBoundingClientRect().bottom) {
+    entry.scrollIntoView();
+  }
+}
+
+// "Connect" button handler.
+async function connect() {
+  let url = document.getElementById('url').value;
+  try {
+    var transport = new WebTransport(url);
+    addToEventLog('Initiating connection...');
+  } catch (e) {
+    addToEventLog('Failed to create connection object. ' + e, 'error');
+    return;
+  }
+
+  try {
+    await transport.ready;
+    addToEventLog('Connection ready.');
+  } catch (e) {
+    addToEventLog('Connection failed. ' + e, 'error');
+    return;
+  }
+
+  transport.closed
+      .then(() => {
+        addToEventLog('Connection closed normally.');
+      })
+      .catch(() => {
+        addToEventLog('Connection closed abruptly.', 'error');
+      });
+
+  globalThis.currentTransport = transport;
+  globalThis.streamNumber = 1;
+  try {
+    globalThis.currentTransportDatagramWriter = transport.datagramWritable.getWriter();
+    addToEventLog('Datagram writer ready.');
+  } catch (e) {
+    addToEventLog('Datagram writer failed. ' + e, 'error');
+    return;
+  }
+  readDatagrams(transport);
+  acceptUnidirectionalStreams(transport);
+  document.forms.sending.elements.send.disabled = false;
+  document.getElementById('connect').disabled = true;
+}
+
+// "Send data" button handler.
+async function sendData() {
+  let form = document.forms.sending.elements;
+  let encoder = new TextEncoder('utf-8');
+  let rawData = sending.data.value;
+  let data = encoder.encode(rawData);
+  let transport = globalThis.currentTransport;
+  try {
+    switch (form.sendtype.value) {
+      case 'datagram':
+        await globalThis.currentTransportDatagramWriter.write(data);
+        addToEventLog('Sent datagram: ' + rawData);
+        break;
+      case 'unidi': {
+        let stream = await transport.createUnidirectionalStream();
+        let writer = stream.writable.getWriter();
+        await writer.write(data);
+        writer.close();
+        addToEventLog('Sent a unidirectional stream with data: ' + rawData);
+        break;
+      }
+      case 'bidi': {
+        let stream = await transport.createBidirectionalStream();
+        let number = globalThis.streamNumber++;
+        readFromIncomingStream(stream, number);
+
+        let writer = stream.writable.getWriter();
+        await writer.write(data);
+        writer.close();
+        addToEventLog(
+            'Opened bidirectional stream #' + number +
+            ' with data: ' + rawData);
+        break;
+      }
+    }
+  } catch (e) {
+    addToEventLog('Error while sending data: ' + e, 'error');
+  }
+}
+// Reads datagrams from |transport| into the event log until EOF is reached.
+async function readDatagrams(transport) {
+  let reader = transport.datagramReadable.getReader();
+  let decoder = new TextDecoder('utf-8');
+  try {
+    while (true) {
+      let result = await reader.read();
```

Comment:
Optional: I prefer to use destructuring assignment like:
```javascript
const { value, done } = await reader.read();
```
but it might be harder to understand?

Reply:
Fixed. Other examples also use destructuring assignment.